focus/operations/src/maintenance/mod.rs

```diff
@@ -242,18 +245,21 @@ impl Runner<'_> {
     }
 
     #[tracing::instrument]
-    fn run_internal_maint(&self, time_period: TimePeriod, repo_path: &Path) -> Result<()> {
-        crate::sync::run(
+    fn run_internal_maint(
+        &self,
+        time_period: TimePeriod,
+        repo_path: &Path,
+    ) -> Result<crate::sync::SyncStatus> {
+        let sync_result = crate::sync::run(
             repo_path,
             crate::sync::SyncMode::Preemptive { force: false },
             self.app.clone(),
         )
-        .with_context(|| format!("Preemptively syncing in {}", repo_path.display()))
-        .map(|_| ())?;
+        .with_context(|| format!("Preemptively syncing in {}", repo_path.display()))?;
 
         self.run_rocksdb_compaction(repo_path)?;
```

Comment:
Do we want to compact regardless of whether the sync succeeds?

Reply:
Oh, I misunderstood what `.map(|_| ())?` was doing here — I read it as `.map_err(...)` i.e. discard the error and run the next line regardless (which would be weird to use with `.with_context`). I think it would make sense to preemptively run RocksDB compaction regardless of my above misunderstanding, but we don't need to touch it in this PR (unless you already made some change to do that).